### PR TITLE
chore: sync go version & cleaning

### PR DESCRIPTION
**Pull request by razbensimon**
### Issue & Steps to Reproduce / Feature Request
[//]: <> (choose “resolves” “fixes” or “closes” and put link for the issue)

### Solution
- sync version of go to 1.16
- replace repo URL in tests

### CODE REVIEW

**Comment by razbensimon** [comment]
[sync CI Golang version with go.mod file](https://trello.com/c/hSKSx0B1/3382-sync-ci-golang-version-with-gomod-file)

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/env0/terraform-provider-env0
 
-go 1.14
+go 1.16
```
Comment (omry-hay) [inline go.mod:3]: Probably you should also add a comment here...

**Review by omry-hay — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (razbensimon) [reply go.mod]: a221df7

Comment (razbensimon) [reply go.mod]: @omry-hay please re-review with 1a80ab6

**Review by omry-hay — APPROVED** [review]

File: tests/integration/004_template/main.tf [hunk 2]
```diff
@@ -6,9 +6,9 @@ resource "env0_template" "tested1" {
   name                                    = "tested1"
   description                             = "Tested 1 description"
   type                                    = "terraform"
-  repository                              = "https://github.com/shlomimatichin/env0-template-jupyter-gpu"
-  github_installation_id                  = 3
-  path                                    = var.second_run ? "second" : ""
+  repository                              = "https://github.com/env0/templates"
+  github_installation_id                  = 11551359
+  path                                    = var.second_run ? "second" : "misc/null-resource"
```
[the remark this answers is not in the document]
Comment (razbensimon) [reply tests/integration/004_template/main.tf:11]: 🤷🏻‍♂️

**Pull request merged**